yapapi/services.py

```diff
@@ -138,18 +140,33 @@ def cluster(self) -> "Cluster":
         return self._cluster
 
     @property
-    def id(self):
+    def id(self) -> str:
         """Return the id of this service instance.
 
         Guaranteed to be unique within a Cluster.
         """
         return self._ctx.id
 
     @property
-    def provider_name(self):
+    def provider_name(self) -> Optional[str]:
         """Return the name of the provider that runs this service instance."""
         return self._ctx.provider_name
 
+    @property
+    def provider_id(self) -> str:
```

Comment:
it's because the underlying properties have those types

Reply:
I don't think this explains anything?

If there is any reason (like a macro-scale reason) to have a Service with defined `provider_id`, but without `provider_name`, this could be documented. If there is no real reason, but this is only an implementation detail, we might note this as a possible candidate for some refactoring. I understand this might not be a part of this PR, but this **is** some sort of a problem.